src/logic/mutinyWalletSetup.ts

```diff
@@ -15,114 +16,109 @@ export type MutinyWalletSettingStrings = {
     scorer?: string;
 };
 
-export function getExistingSettings(): MutinyWalletSettingStrings {
-    const network =
-        localStorage.getItem("MUTINY_SETTINGS_network") ||
-        import.meta.env.VITE_NETWORK;
-    const proxy =
-        localStorage.getItem("MUTINY_SETTINGS_proxy") ||
-        import.meta.env.VITE_PROXY;
-    const esplora =
-        localStorage.getItem("MUTINY_SETTINGS_esplora") ||
-        import.meta.env.VITE_ESPLORA;
-    const rgs =
-        localStorage.getItem("MUTINY_SETTINGS_rgs") || import.meta.env.VITE_RGS;
-    const lsp =
-        localStorage.getItem("MUTINY_SETTINGS_lsp") || import.meta.env.VITE_LSP;
-    const auth =
-        localStorage.getItem("MUTINY_SETTINGS_auth") ||
-        import.meta.env.VITE_AUTH;
-    const subscriptions =
-        localStorage.getItem("MUTINY_SETTINGS_subscriptions") ||
-        import.meta.env.VITE_SUBSCRIPTIONS;
-    const storage =
-        localStorage.getItem("MUTINY_SETTINGS_storage") ||
-        import.meta.env.VITE_STORAGE;
-    const scorer =
-        localStorage.getItem("MUTINY_SETTINGS_scorer") ||
-        import.meta.env.VITE_SCORER;
-
-    return {
-        network,
-        proxy,
-        esplora,
-        rgs,
-        lsp,
-        auth,
-        subscriptions,
-        storage,
-        scorer
-    };
+const SETTINGS_KEYS = [
+    {
+        name: "network",
+        storageKey: "USER_SETTINGS_network",
+        default: import.meta.env.VITE_NETWORK
+    },
+    {
+        name: "proxy",
+        storageKey: "USER_SETTINGS_proxy",
+        default: import.meta.env.VITE_PROXY
+    },
+    {
+        name: "esplora",
+        storageKey: "USER_SETTINGS_esplora",
+        default: import.meta.env.VITE_ESPLORA
+    },
+    {
+        name: "rgs",
+        storageKey: "USER_SETTINGS_rgs",
+        default: import.meta.env.VITE_RGS
+    },
+    {
+        name: "lsp",
+        storageKey: "USER_SETTINGS_lsp",
+        default: import.meta.env.VITE_LSP
+    },
+    {
+        name: "auth",
+        storageKey: "USER_SETTINGS_auth",
+        default: import.meta.env.VITE_AUTH
+    },
+    {
+        name: "subscriptions",
+        storageKey: "USER_SETTINGS_subscriptions",
+        default: import.meta.env.VITE_SUBSCRIPTIONS
+    },
+    {
+        name: "storage",
+        storageKey: "USER_SETTINGS_storage",
+        default: import.meta.env.VITE_STORAGE
+    },
+    {
+        name: "scorer",
+        storageKey: "USER_SETTINGS_scorer",
+        default: import.meta.env.VITE_SCORER
+    }
+];
+
+function getItemOrDefault(
+    storageKey: string,
+    defaultValue: string
+): string | undefined {
+    const item = localStorage.getItem(storageKey);
+    localStorage.ge;
+    if (item === "") {
+        return undefined;
+    } else if (item === null || item === defaultValue) {
+        return defaultValue;
+    } else {
```

Comment:
the `|| item === defaultValue` is redundant because the `else` case will do the same